editoast/src/schema/railjson.rs

```diff
@@ -0,0 +1,74 @@
+use thiserror::Error;
+
+use derivative::Derivative;
+use diesel::PgConnection;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    api_error::ApiError,
+    infra::{Infra, RAILJSON_VERSION},
+};
+
+use super::{
+    BufferStop, Catenary, Detector, OperationalPoint, Route, Signal, SpeedSection, Switch,
+    SwitchType, TrackSection, TrackSectionLink,
+};
+
+#[derive(Deserialize, Derivative, Serialize, Clone, Debug)]
+#[derivative(Default)]
+#[serde(deny_unknown_fields)]
+pub struct RailJson {
+    pub version: String,
+    pub operational_points: Vec<OperationalPoint>,
+    pub routes: Vec<Route>,
+    pub switch_types: Vec<SwitchType>,
+    pub switches: Vec<Switch>,
+    pub track_section_links: Vec<TrackSectionLink>,
+    pub track_sections: Vec<TrackSection>,
+    pub speed_sections: Vec<SpeedSection>,
+    pub catenaries: Vec<Catenary>,
+    pub signals: Vec<Signal>,
+    pub buffer_stops: Vec<BufferStop>,
+    pub detectors: Vec<Detector>,
+}
```

Comment:
Currently you're not using `derivative::Default` so it could be simplified using `Default` for std.
The field `version` could actually have `RAILJSON_VERSION` as default value derivative allows you to do that.

Reply:
ok